### PR TITLE
(fix) Track menu: avoid crash and UX issues with track nullptr

### PR DESCRIPTION
**Pull request by ronso0**
This is just a quick fix to avoid a crash when trying to open the track menu or Track Properties with Ctrl+P
**when the selection contains a duplicate track**. (a track that references the same file as another track)
Note: fixin the UX inconsistencies (see below) is a lot more work, probably requires some not so trivial refactoring.

No idea how I got that duplicate file, but it's awful show stopper.
The general question is how to provide helpful info to the user so they can resolve the situation themselves (without having to edit the database manually).

The menu and Track Properties can now be opened (Track Props is empty) but at least it doesn't crash anymore (in release builds / with DEBUG_ASSERTIONS_FATAL = OFF).

**Note for testing:**
In case you manage to create a duplicate track, you need to disable this DEBUG_ASSERT in TrackDAO inorder to proceed
https://github.com/mixxxdj/mixxx/blob/f5890a89bf1db266cfaaa166e31085aadfe1a1ca/src/library/dao/trackdao.cpp#L1440
FWIW I couldn't duplicate a track with DB Browser for SQlite, it prevents that:
`Error changing data: UNIQUE constraint failed: track_locations.location`

____
Uff, the UX when encountering such a duplicate is very inconsistent, and I think a 'normal' user (DEBUG_ASSERTIONS_FATAL = OFF) does not get enough information to resolve the situation:
1. Track Properties can be opened, but it's empty
2. when opening Track Properties for another track in the table, then hit a dupe track when using Next/Prev, the fields are cleared but the window title is not
3. MusicBrainz dialog can be opened but displays ~20% progress and "Tags can be applied"
4. otoh, the track menu can be opened ~~and most (all?) actions work~~ _was confused due to ¹_
disabled: Find On Web
works: Add to Auto DJ, Add to Playlist/Crate (though sidebar items are not set bold)
not working: BPM, Color, ...
Search Related:
    * is disabled when hovering/selecting it, cursor is stuck there, menu can only be closed by pressing Esc twice
    * remains disabled for regular tracks afterwards (wrong order of populate and setEnabled())
5. when loading the track by dragging it to a deck (loaded by ref/url), the 'original' track¹ is loaded
6. otoh neither double-click in Tracks nor the keyboard shortcuts Shift+Left/Right work (loaded via TrackPointer, but [`WTrackTableView::loadSelectedTrackToGroup`](https://github.com/mixxxdj/mixxx/blob/f5890a89bf1db266cfaaa166e31085aadfe1a1ca/src/widget/wtracktableview.cpp#L1035) aborts due to nullptr)
7. track color is displayed in Tracks but in decks it's the color of the 'original' track?

¹ ~~GlobalTrackCache/TrackDAO throw the warning for the newer track.~~
No, it depends on which track has been cached first. So the dupe can be different in each session.

### CODE REVIEW

**Comment by daschuer** [comment]
does `pTrack = getFirstTrackPointer();` return null? 
I have not yet understand how duplicates are related. But that change is very much required.

**Comment by ronso0** [comment]
TrackDAO returns nullptr when it (actually GlobalTrackCacheResolver) encounters a duplicate.

**Comment by ronso0** [comment]
Uff, the UX when encountering such a duplicate is very inconsistent, and I think a 'normal' user (DEBUG_ASSERTIONS_FATAL = OFF) does not get enough information to resolve the situation:
* Track Properties can be opened, but it's empty
* when opening Track Properties for another track in the table, then hit a dupe track when using Next/Prev, the fields are empty but the window title is not cleared
* otoh the track menu can be opened and most (all?) actions work

I'll make some small fixes soonish.

Note: GlobalTrackCache/TrackDAO throw the warning for the newer track.

**Comment by ronso0** [comment]
Some fixes...

If desired, I can move the DlgTagFetcher fix to a separate PR.

**Review by daschuer — APPROVED** [review]
LGTM. Thank you.  What is your plan with the TODOs?

File: src/widget/wtrackmenu.cpp [hunk 1]
```diff
@@ -964,6 +965,7 @@ void WTrackMenu::updateMenus() {
         bool anyBpmLocked;
         bool anyBpmNotLocked;
         std::tie(anyBpmLocked, anyBpmNotLocked) = getTrackBpmLockStates();
+        // TODO Disable menu if anyBpmLocked == anyBpmNotLocked ??
```
[the remark this answers is not in the document]
Comment (ronso0) [reply src/widget/wtrackmenu.cpp:968]: Hmm, the TODO text is not clear. What I meant was: disable the menu if both are false which happens if getTrackBpmLockStates() found no valid track.

But actually, if users run into null tracks unnoticed, those tracks won't be processed and the log should contain a warning, which IMO is more helpful to discover null tracks than greying out the menu.

I'll remove the TODO

**Review by daschuer — APPROVED** [review]
LGTM, Thank you.

**Pull request merged**